src/coreclr/src/jit/compiler.hpp

```diff
@@ -1301,6 +1301,7 @@ inline GenTree* Compiler::gtNewIndir(var_types typ, GenTree* addr)
 
 inline GenTree* Compiler::gtNewNullCheck(GenTree* addr, BasicBlock* basicBlock)
 {
+    assert(fgAddrCouldBeNull(addr));
```

Comment:
It is questionable if it should be an assert or
```
if (!fgAddrCouldBeNull(addr))
{
  // No need for a null-check.
  return addr;
}
```
or something else.

Reply:
it'd be weird for `gtNewNullCheck` to return something other than null-check 🙂